docs/sources/concepts/configuration-syntax/_index.md

```diff
@@ -84,26 +114,21 @@ The most common expression is to reference the exports of a component, for examp
 You form a reference to a component's exports by merging the component's name (for example, `local.file`),
 label (for example, `password_file`), and export name (for example, `content`), delimited by a period.
 
-## Blocks
-
-You use _Blocks_ to configure components and groups of attributes.
-Each block can contain any number of attributes or nested blocks.
+## Configuration syntax design goals
 
-```alloy
-prometheus.remote_write "default" {
-  endpoint {
-    url = "http://localhost:9009/api/prom/push"
-  }
-}
-```
+{{< param "PRODUCT_NAME" >}} is designed with the following requirements in mind:
 
-The preceding example has two blocks:
+* _Fast_: The configuration language must be fast so the component controller can quickly evaluate changes.
+* _Simple_: The configuration language must be easy to read and write to minimize the learning curve.
+* _Debuggable_: The configuration language must give detailed information when there's a mistake in the configuration file.
 
-* `prometheus.remote_write "default"`: A labeled block which instantiates a `prometheus.remote_write` component.
-  The label is the string `"default"`.
-* `endpoint`: An unlabeled block inside the component that configures an endpoint to send metrics to.
-  This block sets the `url` attribute to specify the endpoint.
+The {{< param "PRODUCT_NAME" >}} configuration syntax is a distinct language with custom syntax and features, such as first-class functions.
 
+* Blocks are a group of related settings and usually represent creating a component.
+  Blocks have a name that consists of zero or more identifiers separated by `.`, an optional user label, and a body containing attributes and nested blocks.
+* Attributes appear within blocks and assign a value to a name.
+* Expressions represent a value, either literally or by referencing and combining other values.
+  You use expressions to compute a value for an attribute.
```

Comment:
Likely not in scope for this PR, but I wonder if this content adds anything new and if it makes more sense to merge it into other headers. 

WDYT about the long-term value of this section @moxious?

Reply:
I was thinking the same. These points feel like a repetition or a rephrase of previous info... and it might be better to find a way to rephrase and fit it into the content around lines 59-65... somewhere in that area.  Might help to commit some of the other changes and do a `make docs` so this can all be read in the rendered output.

Reply:
Noting the goals of the design for a configuration syntax is, in my view - of primary interest to the people who want to hack on the syntax or contribute to core alloy.  So it's pretty advanced material.  That's why I chose to locate it at the bottom.  It wasn't in the scope of this PR to try to remove something, but being frank - if you're a beginning user you probably won't read this, because just knowing how to use the config syntax is enough.

If I were to move this content, I would move it into a section that's about "how to contribute to alloy" or "why we did what we did".  Very useful information to gate what kind of contributions would be welcome.